src/services/infra/DynamoDBService.ts

```diff
@@ -32,10 +35,38 @@ export default class DynamoDBService {
     await this.dynamoDBClient.createItem(this.TABLE_NAME, message)
   }
 
-  async getAllCompletedLaunches(): Promise<SiteLaunchMessage[]> {
+  async getAllLaunches(): Promise<SiteLaunchMessage[]> {
     const entries = ((
       await this.dynamoDBClient.getAllItems(this.TABLE_NAME)
     ).Items?.filter(isSiteLaunchMessage) as unknown) as SiteLaunchMessage[]
+    return entries
+  }
+
+  getLaunchStatus(
+    repoName: string
+  ): ResultAsync<
+    "success" | "failure" | "pending",
+    DatabaseError | MissingSiteError
+  > {
+    return ResultAsync.fromPromise(this.getAllLaunches(), (error) =>
+      createErrorAndLog(
+        DatabaseError,
+        `Something went wrong when querying DynamoDB: ${error}`
+      )
+    ).andThen((entries) => {
+      const entry = entries.find((e) => e.repoName === repoName)
+      if (entry) return okAsync(entry.status?.state ?? "pending")
+      return errAsync(
+        createErrorAndLog(
+          MissingSiteError,
+          `No site found for ${repoName} in DynamoDB`
+        )
```

Comment:
I think we could update the error message to reflect the possible reasons as well (already ended, haven't started)? So this is easier for engineers to debug in future

Reply:
yup is in the roadmap, making more meaningful error messages in a separate pr bah